### DISCUSSION TITLE
How to download a file?

### DISCUSSION THREAD

**Discussion by knightpp** [mymmrac/telego#92] [Q&A] [answered]
There's only https://pkg.go.dev/github.com/mymmrac/telego#Bot.GetFile function related to downloading files. How can I use its `FilePath` field to download a file?

**Comment by mymmrac** [comment, marked as answer]
It's actually written in comment of this func, `The file can then be downloaded via the link https://api.telegram.org/file/bot<token>/<file_path>, where <file_path> is taken from the response.`.

So you would download it from URL using something like this:

```go
DownloadFile("some.file", "https://api.telegram.org/file/bot<token>/<file_path>")

// ...

func DownloadFile(filepath string, url string) error {
	// Get the data
	resp, err := http.Get(url)
	if err != nil {
		return err
	}
	defer resp.Body.Close()

	// Create the file
	out, err := os.Create(filepath)
	if err != nil {
		return err
	}
	defer out.Close()

	// Write the body to file
	_, err = io.Copy(out, resp.Body)
	return err
}
```

> Source: https://golangcode.com/download-a-file-from-a-url

**Reply by knightpp** [reply]
I thought it's weird to use raw TG API. You probably should add a helper function. Thanks, solved.

**Reply by mymmrac** [reply]
Good point I guess I will add it

**Reply by mymmrac** [reply]
@knightpp I added new utility to download files (example [here](https://github.com/mymmrac/telego/blob/main/examples/download_file/main.go))
